examples/next-openai-pages/pages/chat/stream-chat-attachments-url/index.tsx

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable @next/next/no-img-element */
+import { useChat } from 'ai/react';
+import { useRef, useState } from 'react';
+import { upload } from '@vercel/blob/client';
+import { Attachment } from '@ai-sdk/ui-utils';
+
+export default function Page() {
+  const { messages, input, handleSubmit, handleInputChange, isLoading } =
+    useChat({
+      api: '/api/stream-chat',
+    });
+
+  const [attachments, setAttachments] = useState<Attachment[]>([]);
+  const [isUploading, setIsUploading] = useState<boolean>(false);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  return (
+    <div className="flex flex-col gap-2">
+      <div className="flex flex-col p-2 gap-2">
+        {messages.map(message => (
+          <div key={message.id} className="flex flex-row gap-2">
+            <div className="w-24 text-zinc-500 flex-shrink-0">{`${message.role}: `}</div>
+
+            <div className="flex flex-col gap-2">
+              {message.content}
+
+              <div className="flex flex-row gap-2">
+                {message.experimental_attachments?.map((attachment, index) => (
+                  <img
+                    key={`${message.id}-${index}`}
+                    className="w-24 rounded-md"
+                    src={attachment.url}
+                    alt={attachment.name}
+                  />
+                ))}
+              </div>
+            </div>
+          </div>
+        ))}
+      </div>
+
+      <form
+        onSubmit={event => {
+          if (isUploading) {
+            alert('Please wait for the files to finish uploading.');
```

Comment:
can we disable instead?

Reply:
disabling the input would mean the user cannot type any message during upload, so figured an alert works best